### PR TITLE
#65 Fix for JSON Object case

### PR DESCRIPTION
**Pull request by Anatolii**
I found a way to fix the issue for Object case. I think the Array one will be a bit more complex.

### CODE REVIEW

**Comment by vishwakarma** [comment]
I checked the code, the array part should be already covered, although I didn't get chance to test it.

File: src/main/java/com/flipkart/zjsonpatch/InPlaceApplyProcessor.java [hunk 1]
```diff
@@ -147,9 +147,9 @@ public void replace(List<String> path, JsonNode value) {
             String fieldToReplace = path.get(path.size() - 1).replaceAll("\"", "");
             if (isNullOrEmpty(fieldToReplace) && path.size() == 1)
                 target = value;
-            else if (parentNode.isObject())
-                ((ObjectNode) parentNode).put(fieldToReplace, value);
```
Comment (Anatolii) [inline src/main/java/com/flipkart/zjsonpatch/InPlaceApplyProcessor.java]: This method is deprecated now, but internally new 'replace' method does the same: puts a node to the parent. 
I wonder if issue should be also reported in Jackson library?

**Comment by vishwakarma** [comment]
@Anatolii, the code changes looks good.
The Jackson library doesn't state (as per documentation ) that target path should exist for Replace method, but maybe you should try raising this issue on their Github.

Further,  shall we add a backward compatibility flag for the existing users ( while migrating to this version after the code fix)?

**Comment by vishwakarma** [comment]
@Anatolii ping ?

**Comment by Anatolii** [comment]
Sorry, didn't understood that the question about compatibility flag could be addressed to me. 
Since I started using library only recently I don't have any problems with updating to newer version.
As for Jackson bug I didn't have a chance to raise one.
I see that all checks are passed, but I don't have write access to merge it.

**Comment by elfungi** [comment]
@vishwakarma - do you have any plans on adding that backwards-compatibility flag? We would like to pull in the latest updates, but our code is not compatible after this update.

**Pull request merged**